torch/csrc/distributed/rpc/testing/faulty_process_group_agent.h

```diff
@@ -51,6 +51,8 @@ class FaultyProcessGroupAgent : public ProcessGroupAgent {
 
   // Overrides ProcessGroupAgent's enqueueSend to inject delays.
   void enqueueSend(SendWork work) override;
+  // Override ProcessGroupAgent's sendToSelf to inject delays.
+  void sendToSelf(Message&& message) override;
```

Comment:
Any reason this needs to be public?

Reply:
Good point, we can actually make both `sendToSelf` and `enqueueSend` private in this class.